src/tests/unittests/configuration_helper/test_instrument_configuration_set.py

```diff
@@ -0,0 +1,111 @@
+from unittest import TestCase
+
+from unittest.mock import patch, MagicMock, Mock
+
+from qilib.configuration_helper import InstrumentConfiguration
+from qilib.configuration_helper.instrument_configuration_set import InstrumentConfigurationSet, DuplicateTagError
+from qilib.utils.storage import StorageMemory
+
+
+class TestInstrumentConfigurationSet(TestCase):
+    def setUp(self):
+        self._storage = StorageMemory('test_storage')
+
+    def test_constructor(self):
+        instrument_configuration_set = InstrumentConfigurationSet(self._storage)
+        self.assertEqual(instrument_configuration_set.instruments, [])
+        self.assertEqual(len(instrument_configuration_set.tag), 2)
+        self.assertEqual(instrument_configuration_set.tag[0], 'configuration_set')
+        self.assertRegex(instrument_configuration_set.tag[1], r'^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{6}')
+
+    def test_load(self):
+        instrument_configuration_set = InstrumentConfigurationSet(self._storage)
+        instrument_configuration_set.store()
+
+        instrument_configuration_set_new = InstrumentConfigurationSet.load(instrument_configuration_set.tag,
+                                                                           self._storage)
+
+        self.assertEqual(instrument_configuration_set_new.tag, instrument_configuration_set.tag)
+        self.assertEqual(instrument_configuration_set_new.instruments, instrument_configuration_set.instruments)
+
+    def test_store(self):
+        instrument_configuration_set = InstrumentConfigurationSet(self._storage)
+        instrument_configuration_set.store()
+
+        self.assertTrue(self._storage.tag_in_storage(instrument_configuration_set.tag))
+
+    def test_store_with_instruments(self):
+        with patch('qilib.configuration_helper.instrument_configuration.InstrumentAdapterFactory'):
+            instrument_1 = InstrumentConfiguration('DummyClass', 'fake-address-1', self._storage, tag=['instrument_1'])
+            instrument_2 = InstrumentConfiguration('DummyClass', 'fake-address-2', self._storage, tag=['instrument_2'])
+            instrument_1.store()
+            instrument_2.store()
```

Comment:
Since you are testing the `store` on the set I would remove the two lines above.

Reply:
Agree